lib/commands/search.js

```diff
@@ -14,12 +14,10 @@ function search(logger, name, config) {
     tracker = new Tracker(config);
     tracker.track('search', name);
 
-    // If no name was specified, list all packages
-    if (!name) {
-        return Q.nfcall(registryClient.list.bind(registryClient));
-    // Otherwise search it
-    } else {
+    if (name && !config.json) {
         return Q.nfcall(registryClient.search.bind(registryClient), name);
+    } else if (config.json) {
+        return Q.nfcall(registryClient.list.bind(registryClient));
     }
```

Comment:
There used to be comments explaining the logic but now there are not. 

Why does `config.json` need to be present to search all?
Why does `config.json` need to be missing to search by name?
What happens when there is no `config.json` and no `name`?